packages/typegpu/src/types.ts

```diff
@@ -84,26 +125,31 @@ export interface ResolutionCtx {
   };
 }
 
-export interface TgpuResolvable {
-  readonly label?: string | undefined;
-  resolve(ctx: ResolutionCtx): string;
+/**
+ * Houses a method '~resolve` that returns a code string
+ * representing it, as opposed to offloading the resolution
+ * to another mechanism.
+ */
+export interface SelfResolvable {
+  '~resolve'(ctx: ResolutionCtx): string;
   toString(): string;
 }
 
-export function isResolvable(value: unknown): value is TgpuResolvable {
-  return (
-    !!value &&
-    (typeof value === 'object' || typeof value === 'function') &&
-    'resolve' in value
-  );
+export interface Labelled {
+  readonly label?: string | undefined;
```

Comment:
this interface is used exclusively for functions now, but a lot of things have a `label` property :/